packages/experimental/RadioGroup/src/Radio/useRadio.win32.ts

```diff
@@ -51,10 +52,59 @@ export const useRadio = (props: RadioProps): RadioInfo => {
     }
   }, []);
 
+  const isCircularNavigation = true;
+
+  const onInvoke = React.useCallback(
+    (e: KeyPressEvent) => {
+      if (
+        e.nativeEvent.key == 'ArrowDown' ||
+        e.nativeEvent.key == 'ArrowRight' ||
+        e.nativeEvent.key == 'ArrowUp' ||
+        e.nativeEvent.key == 'ArrowLeft'
+      ) {
+        const length = radioGroupContext.enabledValues.length;
+        const currRadioIndex = radioGroupContext.enabledValues.indexOf(radioGroupContext.value);
+        let newCurrRadioIndex;
+        if (e.nativeEvent.key === 'ArrowDown' || e.nativeEvent.key == 'ArrowRight') {
+          if (isCircularNavigation || !(currRadioIndex + 1 == length)) {
+            newCurrRadioIndex = (currRadioIndex + 1) % length;
+            radioGroupContext.value = radioGroupContext.enabledValues[newCurrRadioIndex];
```

Comment:
Are we allowed to explicitly change RadioGroup context values? I'm not fresh on how Context works, but on the two lines below, we're using a callback so the RadioGroup itself can change the values. Here, we're explicitly setting radioGroupContext.value.

Maybe we try making a `changeValue(newVal)` function in the context and call that here?

Again, not sure entirely if this is correct or not, if you think it's ok, we can leave it

Reply:
Other controls, like Tab, have a few instances where they explicitly change context value. However, in this case, the RadioGroup context value is updated with the onChange callback so I removed this line as it is redundant. Thanks for pointing this out!